test/byContract/IexecPoco/01_initialize.test.ts

```diff
@@ -0,0 +1,214 @@
+        };
+    }
+
+    describe('Initialize', function () {
+        it('Should initialize', async function () {
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+            });
+            const { dealId, taskId, taskIndex, startTime } =
+                await iexecWrapper.signAndMatchOrders(orders);
+
+            expect(await iexecPocoAsAnyone.callStatic.initialize(dealId, taskIndex)).to.equal(
+                taskId,
+            );
+            const initialize = await iexecPocoAsAnyone.initialize(dealId, taskIndex);
+            await initialize.wait();
+            await expect(initialize)
+                .to.emit(iexecPoco, 'TaskInitialize')
+                .withArgs(taskId, orders.workerpool.workerpool);
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.status).equal(TaskStatusEnum.ACTIVE);
+            expect(task.dealid).equal(dealId);
+            expect(task.idx).equal(taskIndex);
+            expect(task.timeref).equal(categoryTime);
+            expect(task.contributionDeadline).equal(startTime + 7 * categoryTime);
+            expect(task.finalDeadline).equal(startTime + 10 * categoryTime);
+            // m_consensus does not have any getter
+        });
+
+        it('Should not initialize since index is too low', async function () {
+            const { appOrder, datasetOrder, requestOrder } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+                volume: 10,
+            });
+            const workerpoolOrder0 = {
+                ...createEmptyWorkerpoolOrder(),
+                workerpool: workerpoolAddress,
+                workerpoolPrice,
+                volume: 1,
+            };
+            const workerpoolOrder1 = {
+                ...createEmptyWorkerpoolOrder(),
+                workerpool: workerpoolAddress,
+                workerpoolPrice,
+                volume: 9,
+            };
+            // Request order is matched in 2 deals
+            const { dealId: dealId0 } = await iexecWrapper.signAndMatchOrders({
+                app: appOrder,
+                dataset: datasetOrder,
+                workerpool: workerpoolOrder0,
+                requester: requestOrder,
+            });
+            const { dealId: dealId1 } = await iexecWrapper.signAndMatchOrders({
+                app: appOrder,
+                dataset: datasetOrder,
+                workerpool: workerpoolOrder1,
+                requester: requestOrder,
+            });
+            expect((await iexecPoco.viewDeal(dealId0)).botFirst).equal(0);
+            expect((await iexecPoco.viewDeal(dealId1)).botFirst).equal(1);
+
+            // Will fail since passed taskIndex is below deal1.botFirst
+            await expect(iexecPocoAsAnyone.initialize(dealId1, 0)).to.be.reverted;
```

Comment:
Make the test a bit more accurate, otherwise any revert would pass the test.
```suggestion
            await expect(iexecPocoAsAnyone.initialize(dealId1, 0)).to.be.revertedWithoutReason();
```
With a TODO in Poco2Delegate#initialize() to add messages in require statements.

Reply:
Same for other `.to.be.reverted`.

Reply:
Added the `TODO` :) What is not "accurate"?

Reply:
Thanks!
I mean by using the generic `reverted` any revert would be accepted by the test which could hide regressions.

Reply:
Ok thanks!